src/wallet/wallet.cpp

```diff
@@ -1408,13 +1407,25 @@ bool CWallet::IsHDEnabled()
     return !hdChain.masterKeyID.IsNull();
 }
 
+void CWallet::SetWalletFlag(uint64_t flag)
+{
+    walletFlags |= flag;
```

Comment:
@jonasschnelli the unlogic thing is that [SetWalletFlag](https://github.com/bitcoin/bitcoin/pull/9662#discussion-diff-101948618R1410) switch a bit, when [SetWalletFlags](https://github.com/jonasschnelli/bitcoin/blob/e581d843f06d5fe71a4662b122f94577e5dccd55/src/wallet/wallet.cpp#L1417) set the walletFlags to a specific value.